Minor speedups to movement

vuonojenmustaturska opened the pull request:
Never gonna die in no ghetto

[Changelogs]:

:cl: Naksu
tweak: Minor speedups to movement processing.
change: Fat mobs no longer gain temperature by running.
/:cl:

AnturK committed:
Pirate Event

Buggy123 committed:
clamps view range and add view range to a minimum of 7

Cruix committed:
Fixed teleporting food out of the deep fryer

ExcessiveUseOfCobblestone committed:
hello!

silicons committed:
Adds drill hair

Mark9013100 committed:
Renames charcoal bottles

vuonojenmustaturska committed:
Surviving edged weapons

vuonojenmustaturska committed:
Surviving edged weapons

optimumtact commented:
any proof this is faster

vuonojenmustaturska committed:
Absolutely never

vuonojenmustaturska committed:
Absolutely never

Cruix committed:
Allowed people with telekenesis to eject things from the deep fryer
changed a get_turf() to a drop_location()

vuonojenmustaturska committed:
The fight is on

vuonojenmustaturska committed:
The fight is on

vuonojenmustaturska commented:
/mob/CanPass() is cut down to 25% (proc call memes), also minor speedups to other procs that called checkpass on /atom/movable.

| Proc Name                  | Self CPU | Total CPU | Real Time | Calls |
|----------------------------|----------|-----------|-----------|-------|
| /mob/NewCanPass            | 0.004    | 0.004     | 0.008     | 5882  |
| /mob/OldCanPass            | 0.013    | 0.016     | 0.012     | 6007  |
| /mob/living/carbon/NewMove | 0.003    | 0.006     | 0.005     | 795   |
| /mob/living/carbon/OldMove | 0.004    | 0.009     | 0.005     | 724   |

As for the rest, they call parent in really annoying ways and are hard to A:B in the same session. The changes should be obvious though.

vuonojenmustaturska committed:
oops

vuonojenmustaturska committed:
oops

vuonojenmustaturska committed:
You know what, i'm gonna commit an empty space removal

vuonojenmustaturska committed:
You know what, i'm gonna commit an empty space removal

SpaceManiac committed:
Fix portable turrets attacking some heads and sec

Previously, when set to "non-security and non-command personnel", the
portable turret would only allow the Captain, HoS, Warden, and Security
Officers. Notably, the Detective and other command personnel would be
attacked.

Now, the Detective, Lawyer, and other Heads of Staff are permitted.

Incoming5643 committed:
Doubles the length of Twitterize's accepted string length (#32552)

twitter made a bad decision and I intend to capitalize on it

PKPenguin321 committed:
Grammar fix

YPOQ committed:
Fixes AIs being unable to place robotics factories

Commit:
ok now i kinda tested the ark examine (#32445)

* No, I did NOT test the Ark examine.

* Update countdown.dm

* Update ark_of_the_clockwork_justicar.dm

* Update ark_of_the_clockwork_justicar.dm

* OK JOAN FIXED

* Update ark_of_the_clockwork_justicar.dm

Iamgoofball committed:
jesus christ what the hell

why

Iamgoofball committed:
Update miscellaneous.dm

Qbopper committed:
adjusted default internet song volume to 25

Cyberboss committed:
Merge pull request #32556 from Cruix/deep_fryer

Fixed teleporting food out of the deep fryer

ShizCalev committed:
Adds icons for items with randomization (#32523)

* Adds icons for items with randomization

* Glowsticks

* added ?

* more question marks

* Loc

Cyberboss committed:
Merge pull request #32545 from Buggy123/patch-33

clamps view range and add view range to a minimum of 7

as334 committed:
Assmos tweaks/minor additions. (#32400)

* Adds pluoxium formation, makes fusion radioactive and nerfs it, makes noblium cost more energy.

* Review response.

* tanks now melt above 1 Million degrees Kelvin.

tgstation-server committed:
Automatic changelog generation for PR #32400 [ci skip]

DaxDupont committed:
Fixes medbots not injecting from one tile away

lzimann commented on code/__DEFINES/movement.dm:
```diff
@@ -0,0 +1,2 @@
+// Used to be atom/movable/proc/checkpass
+#define CHECKPASS(mover, flag) (mover.pass_flags & flag)
```
a define for this seems unecessary

lzimann commented on code/modules/mob/living/carbon/carbon_movement.dm:
```diff
@@ -56,8 +56,6 @@
 			nutrition -= HUNGER_FACTOR/10
 			if(m_intent == MOVE_INTENT_RUN)
 				nutrition -= HUNGER_FACTOR/10
-		if((disabilities & FAT) && m_intent == MOVE_INTENT_RUN && bodytemperature <= 360)
```
accidental removal?

vuonojenmustaturska replied:
No, completely intentional. It shortcircuits from the disabilities & FAT check but this proc gets called every time a carbon moves and fat people exist for maybe one minute, once in every 50 rounds since eating is completely optional. So why bother?

I can take it out if it's actually a feature anyone wants in the codebase.

< Naksu> korphaeron: is fat people getting hot when running a feature we really need in /mob/living/carbon/Move
< Naksu> especially given that a round very likely doesn't have fat people in it
<@korphaeron> probably not naksu

lzimann committed:
Merge pull request #32560 from Mark9013100/11/9/17_char_antitox

Renames charcoal bottles

tgstation-server committed:
Automatic changelog generation for PR #32560 [ci skip]

JJRcop committed:
Adds default volume to tgchat javascript as well

MrsTonedOne committed:
SSoverlay improvements and tweaks (#32371)

* Overlay per-type cost logging.

Overlays will now log how long each type took to process.

Changed up how overlays was done to account for the fact its a queue and not a processor. (it was using almost none of the processing subsystem framework)

Made the overlay loop faster by making it not cut the list until the end.

Added a simple generic benchmark stat tracking system.

I don't know how much overhead this adds to overlays. i may put it behind testing or something, but i do want to test this on the serbers to get some stats.

* Removes flush() as it was creating race conditions

* Use ref

* text2file

* Atoms added as an overlay will have their pending overlays compiled before being converted to an appearance

vuonojenmustaturska committed:
removes a define

vuonojenmustaturska committed:
removes a define

vuonojenmustaturska committed:
Remove garbage unused var and a check that can never happen

vuonojenmustaturska committed:
Remove garbage unused var and a check that can never happen

Qbopper committed:
Merge pull request #1 from JJRcop/patch-6

Adds default volume to tgchat javascript as well

tgstation-server committed:
Automatic changelog compile, [ci skip]

AnturK committed:
Fixes readapt removing hivemind communication.

deathride58 committed:
[s] Fixes RPDs qdel'ing girders and causing runtimes (#32567)

* fixes qdeling girders and lattices with RPDs and other weird edge cases

* thanks mom the shitcode's cleaner now

* :+1:

tgstation-server committed:
Automatic changelog generation for PR #32567 [ci skip]

ninjanomnom committed:
Fixes some cases where changeturf were called incorrectly (#32584)

tgstation-server committed:
Automatic changelog generation for PR #32584 [ci skip]

lzimann committed:
Merge pull request #32577 from PKPenguin321/patch-13

Fixes lich revival text spelling

lzimann committed:
Merge pull request #32578 from YPOQ/factoryfix

Fixes AIs being unable to place robotics factories

lzimann committed:
Merge pull request #32574 from AutomaticFrenzy/patch/turret-access

Fix portable turrets attacking some heads and sec

tgstation-server committed:
Automatic changelog generation for PR #32574 [ci skip]

lzimann committed:
Merge pull request #32583 from Iamgoofball/patch-334

Frog masks no longer have a sound

AnturK committed:
Fixes fleeing the station message accuracy

tgstation-server committed:
Automatic changelog generation for PR #32592 [ci skip]

tgstation-server committed:
Automatic changelog compile, [ci skip]

Cyberboss committed:
Updates the clockwork slab description (#32610)

* Updates the clockwork slab description

* Adds period to the end of clockwork slab description

tgstation-server committed:
Automatic changelog generation for PR #32610 [ci skip]

lzimann committed:
Items no longer need to check for tools in attackby, having the check done with the tool_behaviour variable before pre_attackby is called. (#32602)

ninjanomnom committed:
fixes 512 compile reminders (#32582)

AnturK committed:
Makes false alarm use more explicit in event code. (#32559)

tgstation-server committed:
Automatic changelog compile, [ci skip]

ACCount12 committed:
Disposal pipes and pipe dispensers refactor (#32435)

* Disposal pipes and pipe dispensers refactor

* Disposal pipes refactor, phase 2: Reviewing Intensifies

* More code review edits

* Changes hasmob check loop

* There was a point when we should have stopped, and we have clearly passed it.

WJohn committed:
Removes stationary syndicate infiltrator docking ports on all maps. (#32579)

* Removes stationary syndicate infiltrator docking ports on all maps.

* Readds a single stationary docking port for the syndies on all maps just in case their custom docking placement computer is broken. Requested by cruix.

tgstation-server committed:
Automatic changelog generation for PR #32579 [ci skip]

silicons committed:
Fixes Colossus Projectiles (#32621)

YPOQ committed:
Fixes AI anti-cheat false positives (#32627)

lzimann committed:
Merge pull request #32635 from AnturK/cowardsallofthem

Fixes fleeing the station message accuracy

Cyberboss committed:
Merge pull request #32585 from Qbopper/turnitdown

adjusted default internet song volume to 25

tgstation-server committed:
Automatic changelog generation for PR #32585 [ci skip]

nicbn committed:
Space vacuum damage doubled (#32566)

tgstation-server committed:
Automatic changelog generation for PR #32566 [ci skip]

Jalleo committed:
Refactor of base machinery a tiny bit more. DONE (#32546)

* Moves a base uid from machienry to a bunch of children as it was literally setting a var that most machines never used. This is probably a good candidate for a component replacement.

* Just checking something

* Does this fix the compiling not working properly?

* And this?

* What is going wrong ugh!

* params["target"] instead of params["set"] bug was there before I even touched it but my compiling of TGUI caused it or something?

* Slight tidying up for what Cyberboss and Antur likes.

* Fixes the things that I was asked to fix

AnturK committed:
Fixes stimulum decay (#32597)

* Fixes stimulum decay

* Nitryl too

* This too.

* Makes stimulum actually apply it's anti-stun changes.

tortellinitony committed:
Makes clown box consistent with the rest of the boxes (#32642)

silicons committed:
Checks for nulls in weakrefs (#32570)

lzimann committed:
Adds a multitool_act and a proper check on tool_check (#32640)

nicbn committed:
Fixes cryo overlay system  (#32613)

* Fixes cryo overlay system

* oops

SpaceManiac committed:
Prevent placing posters on diagonally-smoothed walls (#32558)

Posters are also knocked off the walls if they change to being
diagonally smoothed later.

tgstation-server committed:
Automatic changelog generation for PR #32558 [ci skip]

Commit:
[WIP] Floyd's wild sound adventure (Adds sound_loop datums to a bunch of machinery) (#32336)

* noisy

* Adds sounds to modular computers

* Adds range modifier var to sound loops

* finetunes computer sound range

* maybe fixes stuff

* changes some of the sound lengths

* borgs and some small changes

* sanity

* gravgen makes noise too

tgstation-server committed:
Automatic changelog generation for PR #32336 [ci skip]

optimumtact committed:
Merge pull request #32557 from kevinz000/hairstyle_drill

Adds Extended Drill Hair

Commit:
Improves the deep fryer (#32482)

While I was messing with vitamins and nutrients, I decided to touch up on deep fryers as a separate project. A demo of sounds can be found here, using the looping sound framework added recently. Also, sound quality in the video is kind of terrible; it sounds much nicer in-game.

Fryers themselves now use the cooking oil reagent, which is new and can be obtained by grinding soybeans (for vegetable oil) or meat (for oil derived from fats.) This does include synthmeat, which you can also use for meat anyway! By default it takes about 1.2 units of oil to fry one object.

Also adds oil_use and fry_speed vars that determine how well the fryer does its job. These values start at ~0.04 and 1 respectively, and can be brought to 0.01 and 4 with max-tier parts (each fryer uses one micro laser.)

Cooking oil itself can be heated to 450 K to bring out its latent frying powers. When splashed on object, it will fry them instantly (albeit inefficiently)... and can also fry humans and other creatures rather painfully.

tgstation-server committed:
Automatic changelog generation for PR #32482 [ci skip]

KorPhaeron committed:
Fixes chaplain not getting his weapon on spawn (#32658)

optimumtact committed:
Merge pull request #32553 from ExcessiveUseOfCobblestone/ihategoofxray

Xrays go through walls but stop on mobs/blob

tgstation-server committed:
Automatic changelog generation for PR #32553 [ci skip]

optimumtact commented:
@MrStonedOne @Jordie0608 @Armhulen @lzimann @KorPhaeron test merge this please and comment when that's done

optimumtact committed:
Merge pull request #31947 from AnturK/yarr

How does this work:
Station receives a communication extorting current cargo point, if they answer yes the points are just gone and nothing of interest happens.

If station refuses to pay or is silent for 3 minutes, pirate shuttle spawns somewhere in space.

    There's an internal GPS onboard so crew will always be able to follow the shuttle.
    Crew of 3, moderately armed. (Balance pending)
    Shuttle engines have 3 minute cooldown between jumps.
    Special shuttle equipment will block cargo and emergency shuttles from leaving and slowly steal the points.

optimumtact committed:
Merge pull request #32618 from AnturK/lingfix

Fixes readapt removing hivemind communication.

praisenarsie committed:
Fixes the broken cuffs (#32667)

tgstation-server committed:
Automatic changelog generation for PR #32667 [ci skip]

nicbn committed:
Adds a timer to falsewall code (#32637)

* Adds a timer to falsewall code

* better code

* removes a parameter

ninjanomnom committed:
makes blueprints work on station and able to expand old areas (#32280)

🆑 ninjanomnom
add: You can now select a nearby area to expand when using a blueprint instead of making a new area.
add: New shuttle areas can be created using blueprints. This is currently not useful but will be used later for shuttle construction.
tweak: Blueprints can modify existing areas on station.
admin: Blueprint functionality has been added to the debug tab as a verb.
/🆑

Additionally the code for getting all turfs in a contained space has been refactored for some pretty good performance improvements.

Old proc: Self CPU 8.816, Total CPU: 10.009, Real Time: 10.009, Calls: 5000
New proc: Self CPU 4.174, Total CPU: 9.000, Real Time: 9.001, Calls: 5000

They've been moved into a area helpers file.

tgstation-server committed:
Automatic changelog generation for PR #32280 [ci skip]

Commit:
Adds ambience for Reebe. (#32623)

* spooky

* fix 1

* fix 2

tgstation-server committed:
Automatic changelog generation for PR #32623 [ci skip]

Jordie0608 commented:
It needs more compile first.

optimumtact commented:
I'm blind as a thot

Jordie0608 commented:
DW it was compiling when you asked before I restarted the build.

vuonojenmustaturska committed:
Rebase on master, fix one new check

vuonojenmustaturska committed:
Merge branch 'movementpopcon' of https://github.com/vuonojenmustaturska/tgstation into movementpopcon

vuonojenmustaturska commented:
oh god

vuonojenmustaturska commented:
semi-unfucked version at #32702
